packages/aws-cdk/lib/api/deployments.ts

```diff
@@ -350,11 +351,11 @@ export class Deployments {
     const cfn = stackSdk.cloudFormation();
 
     // Upload the template, if necessary, before passing it to CFN
-    const cfnParam = await makeBodyParameterAndUpload(
+    const cfnParam = await makeBodyParameter(
       stackArtifact,
       resolvedEnvironment,
+      new AssetManifestBuilder(),
```

Comment:
syntax/style question: `makeBodyParameter` takes an `AssetManifestBuilder()` as an argument - should I make this argument optional so you don't have to pass in a `new AssetManifestBuilder()` like this if you don't want to?

Reply:
I see it [used one other time in deploy-stack.ts here](https://github.com/aws/aws-cdk/blob/b0e4a544aecce86e8b41e7cd148a139c2e34bfbd/packages/aws-cdk/lib/api/deploy-stack.ts#L286) where it looks like we do want to pass in an `AssetManifestBuilder`, so I don't think we'd want to remove the argument entirely

Reply:
I agree that this looks sort of silly. But an `undefined` in the middle there is even sillier. Can we do some other refactor to tease out this useless parameter?

Reply:
We could create another function called `makeBodyParameterFromNewManifest()` (or something like that) which takes all the same arguments as `makeBodyParameter` except for the `AssetManifestBuilder` and just calls `makeBodyParameter` with a new `AssetManifestBuilder`? Like so:

```ts
export async function makeBodyParameterFromNewManifest(
  stack: cxapi.CloudFormationStackArtifact,
  resolvedEnvironment: cxapi.Environment,
  resources: EnvironmentResources,
  sdk: ISDK,
  overrideTemplate?: any,
): Promise<TemplateBodyParameter> {
  return makeBodyParameter(stack, resolvedEnvironment, new AssetManifestBuilder(), resources, sdk, overrideTemplate);
}
```

Not sure if this is also a bit of silly API though.